modules/ocs2/src/main/scala/gem/ocs2/pio/PioParse.scala

```diff
@@ -0,0 +1,42 @@
+package gem.ocs2.pio
+
+import java.time.{Duration, Instant}
+
+import scalaz._
+import Scalaz._
+
+object PioParse {
+  implicit val FunctorPioParse = new Functor[PioParse] {
+    def map[A, B](pa: PioParse[A])(f: A => B): PioParse[B] =
+      pa andThen (_.map(f))
+  }
```

Comment:
This is problematic because `PioParse` is a type alias so `object PioParse` isn't really a companion; the instance isn't in implicit search scope. And `String => ?` is already a functor which is probably going to cause trouble. So I suggest newtyping instead of aliasing:

```scala
case class PioParse[A](run: String => Option[A]) {
  def apply(s: String): Option[A] = run(s) // so we can still treat it like a function
}
```

This will let us define a distinct set of instances (it's isomorphic to `Kleisli[Option, String, A]` so it will have a bunch, but I think `Functor` is probably good enough for now).